src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/BaseOptions.java

```diff
@@ -338,6 +352,34 @@ public boolean process(String opt, List<String> args) {
                     }
                 },
 
+                new XOption(resources, "--date", 1) {
+                    // Valid --date range: within ten years of now
+                    private static final ZonedDateTime now = ZonedDateTime.now();
+                    static final ZonedDateTime DATE_MIN = now.minusYears(10);
+                    static final ZonedDateTime DATE_MAX = now.plusYears(10);
```

Comment:
What's the purpose of limiting valid dates this way?

Reply:
Partly because the code on which I based this on (in the jar tool) had a range check, and partly because I think it makes sense to do a sanity check on the provided value.  The range in the jar tool was 1980-2099 which (see elsewhere) is an artifact of the zip spec, so I changed the range to a "reasonably broad" range of reasonable values.

Given that (for now) we convert internally to a `Date`, we can't accept years before Unix epoch, 1970, anyway, so that's another reason for the check.